### PR TITLE
Add `pre-commit` overview to the dev standards

### PR DESCRIPTION
**Pull request by WillDaSilva**

### CODE REVIEW

**Comment by aaronsteers** [comment]
@WillDaSilva - This is great, thank you for writing this up! 

Is there an 'auto-fix' command we can include here for local fixes? Or does `pipx run pre-commit` _also_ apply fixes? 

And same question for CI workflows. I think pre-commit.ci has an auto-fix command we can put in a comment, but it doesn't match the traditional slash command style, and I can never remember the exact text I need to type to execute it.

**Comment by aaronsteers** [comment]
pre-commit.ci autofix

**Comment by aaronsteers** [comment]
There we go ☝️

**Comment by WillDaSilva** [comment]
> Is there an 'auto-fix' command we can include here for local fixes? Or does pipx run pre-commit also apply fixes?

@aaronsteers `pre-commit` doesn't have special handling for auto-fixes because it merely runs the underlying tools. Ruff, pyupgrade, isort, Black, etc. are what make the changes to your files. A `pre-commit` check will fail if any files are changed while it is running.

> And same question for CI workflows. I think pre-commit.ci has an auto-fix command we can put in a comment [...]

The `pre-commit.ci` GitHub app will run `pre-commit run --all-files` on each PR in the repositories it has access to. The `ci` section of `.pre-commit-config.yaml` can configure with `autofix_prs: true` to commit back any changes that occur while running `pre-commit run --all-files` in CI.

**Comment by aaronsteers** [comment]
Got it - makes sense. I think the one change I'd request would be the instructions on how to automatically run pre-commit from within a PR using the (above) method of typing `'pre-commit.ci autofix'` into a PR.

**Comment by aaronsteers** [comment]
Regarding auto-fixing PRs without requiring the command, I think we should gather broad concensus before turning that on, since (as previously discussed), it can create merge conflicts for the developer and could be somewhat obnoxious if not expected.

Related: do you know if autofix from pre-commit.ci also works on forks?

**Comment by aaronsteers** [comment]
@WillDaSilva - Thanks for catching me up on that. My message re: conflicts were relative to if the commit goes back to the same PR, versus the commit being created in a new PR.

I've seen this run on some repos and I'm happy with the flow I've seen. No blockers, except it would be nice to have a bit more about "what to expect" in this docs update too.

File: src/_engineering/dev-standards.md [hunk 1]
```diff
@@ -22,6 +22,37 @@ Our Meltano value of ongoing `Iteration` is balanced by a requirement that each
...
+
+Our primary reasons for using `pre-commit` are as follows:
+
+- It lets us put our linting dependencies in `.pre-commit-config.yaml` instead of the dev dependency section of `pyproject.toml`. This prevents the transitive dependency restrictions from our linting dependencies from impacting the runtime dependencies. For example, if the latest version of one of our linting dependencies requires `importlib-resources<4.0.0`, but one of our runtime dependencies requires `importlib-resources>=5.0.0`, then we'd likely have to downgrade that runtime dependency until we found a compatible version. Thanks to `pre-commit` managing these dependencies, this is no longer an issue, and we can run `poetry lock` with less fear.
+- It provides (mostly) reproducible isolated environments for linting tools. If a `pre-commit` check is failing in CI, it's probably failing locally too.
+- It has simple CI integration with auto-fixes and auto-updates via [`pre-commit.ci`](https://pre-commit.ci). The `pre-commit.ci` GitHub app is installed in the Meltano and MeltanoLabs GitHub organizations, and is given access on a per-repository basis.
```
Comment (WillDaSilva) [inline src/_engineering/dev-standards.md:35]:
```suggestion
- It has simple CI integration with auto-fixes and auto-updates via [`pre-commit.ci`](https://pre-commit.ci). The `pre-commit.ci` GitHub app is installed in the Meltano and MeltanoLabs GitHub organizations, and is given access on a per-repository basis. If CI autofixes are enabled within `.pre-commit-config.yaml`, then the `pre-commit.ci` application will commit whatever changes result from it running the `pre-commit` checks on all files, if any, to the PR it ran the checks on.
```

**Review by aaronsteers — COMMENTED** [review]
I think this is ready to go! Thanks again, @WillDaSilva, for writing this up and responding to all my questions and feedback. 🙏

**Review by aaronsteers — APPROVED** [review]

**Pull request merged**